### DISCUSSION TITLE
Connecting the srsRAN's  containerized OSC RIC to ns-ORAN

### DISCUSSION THREAD

**Discussion by aliasadi2001** [srsran/srsRAN_Project#578] [General Help]
Greetings all,

I'm seeking assistance in connecting the [Containerized OSC RIC](https://github.com/srsran/oran-sc-ric) with ns-ORAN. Unfortunately, I'm encountering a hurdle as I'm unable to modify ns-ORAN's e2term destination port number, which currently stands at 36422, while the standard port for this connection is 36421. Moreover, the e2term container is set to use port 38000 for connecting to the gNBs. Could someone kindly guide me on how to reconfigure the e2term container to expose port 36422?

Regards

**Comment by pgawlowicz** [comment]
Set the port here: https://github.com/srsran/oran-sc-ric/blob/main/.env#L18 and rebuild docker images

**Reply by aliasadi2001** [reply]
Thank you for your response. Despite updating the port number as instructed and rebuilding the system, the connection issue persists.

This image indicate that the ns-ORAN request terminal operates on the IP address 10.0.2.10 with port 36422. 
![Screenshot from 2024-04-24 18-03-32](https://github.com/srsran/srsRAN_Project/assets/137979233/54383d97-442f-4905-a12a-ea55994f82c9)

Additionally, this is information about the OSC RIC terminal and e2term after port number editing

According to the configuration files, it appears that the OSC RIC is configured to work on the same IP address, 10.0.2.10.
![Screenshot from 2024-04-24 18-07-36](https://github.com/srsran/srsRAN_Project/assets/137979233/911fc34d-374a-452e-9f89-83d41cfd838e)


The most recent version of the Open RAN Gym relies on the bronze release of OSC [RIC.](https://openrangym.com/tutorials/ns-o-ran) My goal is to establish connection with this containerized RIC as an alternative, but so far, I haven't been successful.

I appreciate any assistance you can provide in troubleshooting this issue.

**Reply by pgawlowicz** [reply]
Check if any e2ap messages go over the bridge interface connecting ORAN RIC docker containers.

**Reply by aliasadi2001** [reply]
Hi

Upon reviewing the messages exchanged between bronze RIC and ns-ORAN, I've noticed that there are E2AP messages between them in SCTP format. You can view the screenshot of the conversation here.
![Screenshot from 2024-04-27 16-22-24](https://github.com/srsran/srsRAN_Project/assets/137979233/c82d5fc1-ca13-4d55-97b6-61c54a88096f)


Furthermore, after examining all messages in Wireshark, I've observed that there are no messages between I release OSC RIC and ns-ORAN either. It's worth noting that the I release utilizes E2AP V.3.0.0, whereas the bronze release employs E2AP V.1.0.0. Are E2AP versions backward compatible?

**Reply by pgawlowicz** [reply]
Which wireshark version do you use? You should have at least 4.0.7 or 4.1.0

**Comment by aliasadi2001** [comment]
I've upgraded my Wireshark to version 4.2.4, and I've captured a .pcap file while ns-ORAN and OSC RIC are running.
this is the capturing file in .csv format
[ns-oran.csv](https://github.com/srsran/srsRAN_Project/files/15138430/ns-oran.csv)
there are no sctp message to decode as E2AP.

**Reply by pgawlowicz** [reply]
In your previous wireshark screenshot there was a packet from IP 10.0.2.1 to 10.0.2.10 decoded as X2AP,  I guess it was the E2AP communication.

**Comment by aliasadi2001** [comment]
Thank you .
However, based on these two screenshots, it appears that ns-ORAN is not connected to OSC RIC.

OSC RIC:
![Screenshot from 2024-04-27 18-06-26](https://github.com/srsran/srsRAN_Project/assets/137979233/cc36f417-6e7c-4f07-9eb6-7db00d8eaa09)

ns-ORAN :
![Screenshot from 2024-04-27 18-06-40](https://github.com/srsran/srsRAN_Project/assets/137979233/600d19c2-c8de-4dae-9285-a229d038fd53)

**Comment by pgawlowicz** [comment]
Hmm, in the first pcap we have seen that there was a message from 10.0.2.1 (ns3) to 10.0.2.10 (e2term). Maybe, indeed, the bronze release is not compatible with the release-i.

**Reply by pgawlowicz** [reply]
hi @aliasadi2001, any update on this issue?

**Comment by aliasadi2001** [comment]
Hello,
Unfortunately, I couldn't connect ns-ORAN to containerized OSC RIC.
 I am seeking any help for this connection. 
Thank you.

**Reply by pgawlowicz** [reply]
Probably you would need to update the agent used in ns3 to a newer release.

**Comment by yagoda** [comment]
Hi @aliasadi2001, 

In preparation for our upcoming workshop in DC, we are contacting some users of our E2 solution. Can you please share a few lines about your experience with the E2 interface, what you were trying to achieve, and how it worked out for you?

Kind Regards,
Justin
